policy-controller/k8s/status/src/http_route.rs

```diff
@@ -55,3 +84,158 @@ impl ParentReference {
         }
     }
 }
+
+impl BackendReference {
+    fn from_backend_ref(
+        backend_ref: gateway::BackendObjectReference,
+        default_namespace: &str,
+    ) -> Self {
+        if policy::httproute::backend_ref_targets_kind::<linkerd_policy_controller_k8s_api::Service>(
+            &backend_ref,
+        ) {
+            let namespace = backend_ref
+                .namespace
+                .unwrap_or_else(|| default_namespace.to_string());
+            BackendReference::Service(ResourceId::new(namespace, backend_ref.name))
+        } else {
+            BackendReference::Unknown
+        }
+    }
+}
+
+#[cfg(test)]
+mod test {
+    use super::*;
+    use linkerd_policy_controller_k8s_api::{policy, ObjectMeta};
+
+    fn mk_default_http_backends(
+        backend_refs: Vec<gateway::BackendObjectReference>,
+    ) -> Option<Vec<gateway::HttpBackendRef>> {
+        Some(
+            backend_refs
+                .into_iter()
+                .map(|backend_ref| gateway::HttpBackendRef {
+                    backend_ref: Some(gateway::BackendRef {
+                        inner: backend_ref,
+                        weight: None,
+                    }),
+                    filters: None,
+                })
+                .collect(),
+        )
+    }
+
+    #[test]
+    fn test_backendrefs_from_route() {
```

Comment:
tiny style nit: typically we don't prefix the name of tests with `test_`
```suggestion
    fn backendrefs_from_route() {
```